### ISSUE TITLE
Issue running test_mesh.py testing cube_ascii.stl

### ISSUE THREAD

**Issue by tetov** [compas-dev/compas#360]
**Describe the bug**
Started having a test fail in [`test_mesh.py` , line 74](https://github.com/compas-dev/compas/blob/4e12566473f7479d2ac99ec6d65fa44b7673f168/tests/compas/datastructures/test_mesh.py#L56), while working on PR #348. 

[The mesh in question](https://github.com/compas-dev/compas/blob/master/data/cube_ascii.stl) should have 4020 vertices but I'm getting 4008.

While trying to figure out what went wrong I discovered that fresh installs of both current COMPAS 0.8.1 as well as 0.7.2 reports `mesh.number_of_vertices()` to be 4008. This has been tested both in `virtualenv` and `conda` on more than one machine.

Blender agrees with `test_mesh.py` reporting 4020 vertices while Rhino reports the following:
**TLDR:** non manifold, intersections and 24048 vertices.
```
General information about this mesh:

Mesh has 660 non manifold edges.
Skipping face direction check because of positive non manifold edge count.
Mesh has 10709 pairs of faces that intersect each other.
  This can cause problems if you're doing mesh boolean operations with it.

Mesh does not have any degenerate faces.
Mesh does not have any ngons.
Mesh does not have any extremely short edges.
Mesh does not have any naked edges.
Mesh does not have any duplicate faces.
Mesh has 172 disjoint pieces.
Mesh does not have any unused vertices.


ID: 893f94b4-db11-4879-abee-e38cdb9a4af1 (728)
Object name: (not named)
Layer name: Layer 01
Render Material: 
  source = from layer
  index = -1

Geometry:
  Valid mesh.
  Closed double precision polygon mesh: 24048 vertices, 8016 faces with normals
    Bounding box: (-71.1734,-42.8274,99.5188) to (77.8611,63.5517,177.751)
```

The tests runs fine outside my branch, but I really don't understand why there's a discrepancy in `number_of_vertices()` during `invoke test` and manual testing..

Has anyone got any ideas what's happening here?

**Comment by tetov** [comment]
Paging @Licini since you wrote the tests :). Any thoughts on this specific mesh?

**Comment by tomvanmele** [comment]
it is a precision thing.

```python
import compas
compas.set_precision(0.0000001)
mesh = Mesh.from_stl(compas.get('cube_ascii.stl'))
v = mesh.number_of_vertices()
print(v)
```

finds 4020...

**Comment by tomvanmele** [comment]
perhaps the STL reader should determine the required precision for the parser by checking the number of digits in the X coordinate of the first vertex...

**Comment by tetov** [comment]
Ok! Thanks!

I can add that for ascii in PR #348 and for binary in future PR. 

So glad to have an answer on this, it did doubt my sanity there for a while.

**Comment by tetov** [comment]
Closed due to @brgcode answer, and discussion regarding inferring precision moved to #362.
